### ISSUE TITLE
Completition for type name?

### ISSUE THREAD

**Issue by davemilter** [rust-analyzer/rust-analyzer#3418]
Consider such code:
```rust
enum Enum {
    A,
    B,
}

struct Foo {
    f: Option<Enum>,
}

fn f(foo: &Foo) {
    match foo.f {
        None => (),
        Some(E) => (),
    }
}
```

For some strange reason rust-analyzer can not complete such expression  `Some(E<CURSOR>) => (),` . Strange because of it knows exactly what type inside `Some`.

rust-analyzer knows that this is `Option<Enum>`

![known_type](https://user-images.githubusercontent.com/26097083/75773593-42aafd80-5d5f-11ea-8f49-e0d693e775e6.png)

it knows that inside `Some` there is `Enum`, but there are no completition:
![right_hint](https://user-images.githubusercontent.com/26097083/75773623-50f91980-5d5f-11ea-9622-c55f71f5d4d7.png)

And only when I enter `Enum::` it can complete:

![completition](https://user-images.githubusercontent.com/26097083/75773716-7f76f480-5d5f-11ea-8bd2-9af2817a1cda.png)

**Comment by flodiebold** [comment]
I'm pretty sure there's no completion there because as long as you're just writing `En`, it's just a bind pattern, and you could put any variable name there. We'll need some special handling for this case.

**Comment by matklad** [comment]
The "usual" code path for completion is triggerd here: https://github.com/rust-analyzer/rust-analyzer/blob/9bc9e12cc52ed7dbefa9812133d1902b4bd983dc/crates/ra_ide/src/completion.rs#L69. I think inside `complete_scope` function we should tweak `!is_trivial_path` condition to also handle a case of "a bind pattern, which might be a path". This might need some changes to `CompletionContext` to detect this situation

**Comment by slyngbaek** [comment]
Attempted a pull request here, not sure if I went in the right direction.

**Comment by DianaNites** [comment]
Should this be closed, did #3418 fix this? Was looking looking `E-mentor` issues
